File: form-ajout-util.js
function isItGood(){

        var login = document.getElementById("login");

        if (login.value == ""){
                alert("Erreur: Le login est requis.")
                return false
        }
	
	var mdp = document.getElementById("motdepasse");
	
	if (mdp.value == ""){
		alert("Erreur : Le mot de passe doit être renseigné")
		return false
	}else if (mdp.value.length <= 8 ){
	        alert("Erreur: Le mot de passe n'est pas assez long.")
                return false;
	}

	var conf = document.getElementById("conf");

        if (conf.value != mdp.value){
                alert("Erreur: Les mots de passe sont différents.")
                return false
        }
}

function alreadyUsed(){
	var login = document.getElementById("login");
	var url = "check-login.py?login=" + login.value;

    	var req = new XMLHttpRequest();
    	req.open("GET", url, true);

    	req.onreadystatechange = function(){
        	if(req.readyState == 4 && req.status == 200) {
			var used=document.getElementById("used") ;
                        used.innerHTML = req.responseText;
        	}

    	}
req.send();
}

File: form-ajout-util.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

from mod_python import Session
from fonctions import redirectionSiNonConnecte, lien, codeHTML, connexionBD

def index(req):
    req.content_type="text/html"

    sess = Session.Session(req) #recup session
    redirectionSiNonConnecte(req,sess) #redirige si la session est nouvelle
    id_util=sess["id_util"] #recup l'id_util

#start sql check que l'util est root
    conn = connexionBD()
    cur = conn.cursor()
    sql = "select login from util where util.id_util={} and util.login='{}';".format(id_util, 'root')
    cur.execute(sql)
    conn.commit()
    data = cur.fetchall()
    if not data:
	 req.write(codeHTML("Erreur !", """
<h3><g>Seul l'utilisateur root a le droit d'ajouter un utilisateur.</g></h3>
""" + lien('menu.py',"Retour au menu")))
#end sql check util root

    else:
	req.write(codeHTML("Ajout d'un utilisateur","""
<p><b>Ajout d'un utilisateur</b></p>
<form method="POST" action="ajout-util.py" onsubmit="return isItGood()">
    <table>
        <tr>
            <td>Login</td>
            <td><input type="text" name="login" id="login" onkeyup="alreadyUsed()"/></td>
            <td></td>
        </tr>
        <tr>
            <td>Mot de passe</td>
            <td><input type="password" name="motdepasse" id="motdepasse"/></td>
            <td></td>
        </tr>
        <tr>
            <td>Confirmation du mot de passe</td>
            <td><input type="password" name="conf" id="conf"/></td>
            <td></td>
        </tr>
    </table>
    <div id="used"></div>
</form>
""" + lien("menu.py","Retour au menu principal") + """
<script src="form-ajout-util.js"></script>
"""))
